services/src/main/java/uk/gov/moj/sdt/services/messaging/MessageWriter.java

```diff
@@ -1,188 +1,174 @@
+ * and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice, this list of
+ * conditions and the following disclaimer in the documentation and/or other materials
+ * provided with the distribution.
+ * - All advertising materials mentioning features or use of this software must display the
+ * following acknowledgment: "This product includes Money Claims OnLine."
+ * - Products derived from this software may not be called "Money Claims OnLine" nor may
+ * "Money Claims OnLine" appear in their names without prior written permission of the
+ * Ministry of Justice.
+ * - Redistributions of any form whatsoever must retain the following acknowledgment: "This
+ * product includes Money Claims OnLine."
+ * This software is provided "as is" and any expressed or implied warranties, including, but
+ * not limited to, the implied warranties of merchantability and fitness for a particular purpose are
+ * disclaimed. In no event shall the Ministry of Justice or its contributors be liable for any
+ * direct, indirect, incidental, special, exemplary, or consequential damages (including, but
+ * not limited to, procurement of substitute goods or services; loss of use, data, or profits;
+ * or business interruption). However caused any on any theory of liability, whether in contract,
+ * strict liability, or tort (including negligence or otherwise) arising in any way out of the use of this
+ * software, even if advised of the possibility of such damage.
+ *
+ * $Id: $
+ * $LastChangedRevision: $
+ * $LastChangedDate: $
+ * $LastChangedBy: $ */
+
+package uk.gov.moj.sdt.services.messaging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.moj.sdt.services.messaging.api.IMessageWriter;
+import uk.gov.moj.sdt.services.messaging.api.ISdtMessage;
+import uk.gov.moj.sdt.services.messaging.asb.MessageSender;
+import uk.gov.moj.sdt.utils.SdtContext;
+import uk.gov.moj.sdt.utils.logging.PerformanceLogger;
+import uk.gov.moj.sdt.utils.mbeans.SdtMetricsMBean;
+
+import java.util.GregorianCalendar;
+import java.util.Map;
+
+/**
+ * Message writer that handles writing messages to a message queue.
+ *
+ * @author Manoj Kulkarni
+ */
+@Component
+public class MessageWriter implements IMessageWriter {
+    /**
+     * Logger object.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(MessageWriter.class);
+
+    /**
+     * Default suffix for DLQ name.
+     */
+    private static final String DLQ_SUFFIX = ".DLQ";
+
+    /**
+     * The last id used for a queued message.
+     */
+    private static long lastQueueId;
+
+    /**
+     * The JmsTemplate object from Spring framework.
+     */
+    private final MessageSender messageSender;
+
+    private final QueueConfig queueConfig;
```

Comment:
Should this have a comment describing what this is?  All the other attributes have one.

Reply:
Comments are not really needed